### PR TITLE
Jvenstad/deployment api expansion continued

### PR DESCRIPTION
**Pull request by jonmv**
@bratseth please review.

### CODE REVIEW

**Comment by jonmv** [comment]
I'll do a cleanup and see it JobList can be used to replace existing code, as well, in a different PR. (As well as look for other clean-up opportunities :p)

File: controller-server/src/main/java/com/yahoo/vespa/hosted/controller/restapi/deployment/DeploymentApiHandler.java [hunk 1]
```diff
@@ -132,4 +150,41 @@ private static String toString(DeploymentSpec.UpgradePolicy upgradePolicy) {
...
+
+    /** The last triggered upgrade to this version, for this application */
+    private Optional<JobStatus> lastDeployingTo(Version version, Application application) {
+        return JobList.from(application)
+                .running(controller.applications().deploymentTrigger().jobTimeoutLimit())
+                .lastTriggered().upgrade()
+                .asList().stream()
+                .max(comparing(job -> job.lastTriggered().get().at()));
+    }
+
```
Comment (jonmv) [inline controller-server/src/main/java/com/yahoo/vespa/hosted/controller/restapi/deployment/DeploymentApiHandler.java]: I'll try to merge as much as possible of this with what it should mirror, from VersionStatus, and put it in JobList. ... In a new PR.

**Comment by bratseth** [comment]
Great

**Comment by jonmv** [comment]
The stuff outside JobList and the DeploymentHandler is a revert-revert, so you've seen it before.

**Pull request merged**